File: 백준/Bronze/2581. 소수/소수.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.List;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(br.readLine());
        int m = Integer.parseInt(br.readLine());

        int sum = 0;

        List<Integer> list = new ArrayList<>();

        // N이상 M 이하의 자연수 중 소수인 것 찾기
        for (int i = n; i <= m; i++) {
            if (isPrime(i)) {
                list.add(i);
                sum += i;
            }
        }
        if (sum == 0) {
            System.out.println(-1);
        } else {
            System.out.println(sum);
            System.out.println(list.get(0));
        }
    }

    public static boolean isPrime(int num) {
        if (num <= 1) return false;
        for (int i = 2; i < num; i++) {
            if (num % i == 0) return false;
        }
        return true;
    }
}

File: 백준/Bronze/2581. 소수/소수.c
#include <stdio.h>
#include <stdlib.h>

int isPrime(int num) {
    if (num <= 1) return 0; 
    for (int i = 2; i < num; i++) {
        if (num % i == 0) return 0;
    }
    return 1;
}

int main() {
    int n, m;
    scanf("%d", &n);
    scanf("%d", &m);

    int sum = 0;
    int firstPrime = -1;
    int foundPrime = 0;

    for (int i = n; i <= m; i++) {
        if (isPrime(i)) {
            if (!foundPrime) {
                firstPrime = i;
                foundPrime = 1;
            }
            sum += i;
        }
    }

    if (sum == 0) {
        printf("-1\n");
    } else {
        printf("%d\n", sum);
        printf("%d\n", firstPrime);
    }

    return 0;
}
